pkg/rbac/sharedData.go

```diff
@@ -38,6 +39,13 @@ type SharedData struct {
 	nsErr       error      // Capture errors retrieving namespaces.
 	nsLock      sync.Mutex // Locks the namespaces array while updating it.
 	nsUpdatedAt time.Time  // Time when namespaces data was last updated.
+
+	propTypeErr  error
+	propTypeTime time.Time
+}
+
+type PropTypesCache struct {
+	propType map[string]string //list of all properties and thier relative data types among all users.
 }
```

Comment:
- Why is this needed instead of declaring `propType map[string]string` directly into sharedData?
- If we are declaring a new struct, then I'd combine the error and type like this.
    ```go
	type PropTypesCache struct {
		propType map[string]string //list of all properties and thier relative data types among all users.
		propTypeErr  error
		propTypeTime time.Time
	}
    ```

While both alternatives work, try to keep consistency with the existing code. Or consider refactoring existing code to keep consistency.

Reply:
I guess we don't need to keep it separate from main cache. I was trying to be consistent with userDataCache and how we store the resources in a separate struct for easy access.